kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+	            		} else {
+	            			bankOffsetAmount = getPurchasingAccountsPayableModuleService().getTotalPreTaxDollarAmount(document.getDocumentHeader().getDocumentNumber());
+	            		}
+	            	}
+	            }
+            }
+            
+            if(bankOffsetAmount == null) {
+                bankOffsetAmount = getGeneralLedgerPendingEntryService().getOffsetToCashAmount(document).negated();
+            }
+        }
+        if ( !KualiDecimal.ZERO.equals(bankOffsetAmount) ) {
+            GeneralLedgerPendingEntry bankOffsetEntry = new GeneralLedgerPendingEntry();
+            success &= getGeneralLedgerPendingEntryService()
+                    .populateBankOffsetGeneralLedgerPendingEntry(bank, bankOffsetAmount, document, 
+                            document.getPostingYear(), sequenceHelper, bankOffsetEntry, bankCodePropertyName);
+    
+            if (success) {
+                AccountingDocumentRuleHelperService accountingDocumentRuleUtil = SpringContext.getBean(AccountingDocumentRuleHelperService.class);
+                bankOffsetEntry.setTransactionLedgerEntryDescription(accountingDocumentRuleUtil.formatProperty(KFSKeyConstants.Bank.DESCRIPTION_GLPE_BANK_OFFSET));
+                bankOffsetEntry.setFinancialDocumentTypeCode(documentTypeCode);
+                document.addPendingEntry(bankOffsetEntry);
+                sequenceHelper.increment();
+    
+                GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(bankOffsetEntry);
+                success &= getGeneralLedgerPendingEntryService().populateOffsetGeneralLedgerPendingEntry(document.getPostingYear(), bankOffsetEntry, sequenceHelper, offsetEntry);
+                bankOffsetEntry.setFinancialDocumentTypeCode(documentTypeCode);
+                document.addPendingEntry(offsetEntry);
+                sequenceHelper.increment();
+            }
+        }
+
+        return success;
+    }
+    
+    protected GeneralLedgerPendingEntryService getGeneralLedgerPendingEntryService() {
+        if ( generalLedgerPendingEntryService == null ) {
+            generalLedgerPendingEntryService = SpringContext.getBean(GeneralLedgerPendingEntryService.class);
+        }
+        return generalLedgerPendingEntryService;
+    }
+    
+    public void setGeneralLedgerPendingEntryService(GeneralLedgerPendingEntryService generalLedgerPendingEntryService) {
+    	this.generalLedgerPendingEntryService = generalLedgerPendingEntryService;
+    }
+    
+    protected ObjectCodeService getObjectCodeService() {
+        if ( objectCodeService == null ) {
+            objectCodeService = SpringContext.getBean(ObjectCodeService.class);
+        }
+        return objectCodeService;
+    }
+    
+    public void setObjectCodeService(ObjectCodeService objectCodeService) {
+    	this.objectCodeService = objectCodeService;
+    }
+    
+    protected ParameterService getParameterService() {
```

Comment:
getParameterService() is not actually being called. Since the service is being injected, only the setter is needed.